tests/micro/stm32/test_code_emitter.py

```diff
@@ -0,0 +1,390 @@
+#   check_network
+# ========================================================
+def check_network(build_dir, target_name, model_path, image_path):
+
+    model_name = "network"
+
+    model, shape_dict, dtype_dict = get_tflite_model(model_path)
+
+    #
+    # Generate random input data
+    #
+    image_data = []
+    for i in range(NUM_ITERATIONS):
+        assert len(shape_dict) == 1, "Only single input models are supported."
+        image_shape = list(shape_dict.values())[0]
+        in_data = np.random.randint(0, 255, size=image_shape).astype("uint8")
+        # Write raw data for using with the TVM implementation
+        filename = os.path.join(image_path, "{:02d}.raw".format(i))
+        dump_image(filename, in_data)
+        image_data.append(in_data)
+
+    mod, params = relay.frontend.from_tflite(model, shape_dict, dtype_dict)
+
+    #
+    # Build a TVM C module for the ARM CPU (without compiling the kernels
+    # library to the object code form):
+    #
+    with tvm.transform.PassContext(opt_level=3, config={"tir.disable_vectorize": True}):
+        rt_module = relay.build(mod, target="c -device=arm_cpu", params=params)
+
+    #
+    # Export model library format
+    #
+    target_dir = os.path.join(build_dir, target_name + "_gen")
+
+    if os.path.exists(target_dir):
+        print(f'Removing existing "{target_dir}" directory')
+        try:
+            shutil.rmtree(target_dir)
+        except OSError as err:
+            raise ValueError(f"emit_code.Error: {target_dir} : {err.strerror}")
+
+    mlf_tar_path = os.path.join(build_dir, target_name + "_lib.tar")
+    import tvm.micro as micro
+
+    micro.export_model_library_format(rt_module, mlf_tar_path)
+
+    emitter = stm32.CodeEmitter()
+    quantization = extract_tflite_quantization(model)
+    emitter.parse_library_format(mlf_tar_path, quantization)
+    emitter.emit_code(target_dir, model_name)
+
+    #
+    # Results
+    #
+    tf_results = run_tflite_model(model_path, image_data)
+    tvm_results = run_tvm_model(build_dir, model_name, target_dir, image_path)
+
+    check_result(tf_results, tvm_results)
+
+
+# ========================================================
+#   check_result
+# ========================================================
+def check_result(tflite_results, tvm_results):
+    """Helper function to verify results"""
+
+    #
+    # MNIST quantized uint8 results in one single difference of
+    # ~ 0.004 so just escape this
+    #
+    ATOL = 1e-3
+    RTOL = 0.5
+
+    tvm.testing.assert_allclose(tflite_results, tvm_results, rtol=RTOL, atol=ATOL)
+
+
+# ========================================================
+#   test_mnist
+# ========================================================
+def test_mnist():
+    curr_path = os.path.dirname(os.path.abspath(os.path.expanduser(__file__)))
+    model_path = os.path.join(curr_path, "models/mnist.tflite")
+    build_dir = os.path.join(curr_path, BUILD_DIR)
```

Comment:
as we add more tests, we'll need to config filesystem modifications in a test-specific directory. can you do this:
```
DEBUG = False
tempdir_root = None
if DEBUG:
  tempdir_root = os.path.join(curr_path, f"workspace", "test_mnist", datetime.datetime.now().strftime("%Y-%m-%dT%H-%M-%S"))
build_dir = tvm.contrib.utils.tempdir(tempdir_root)
```